### DISCUSSION TITLE
How to use img2img?

### DISCUSSION THREAD

**Discussion by Deadstarrr** [hlky/stable-diffusion#263] [Q&A]
When I add an image and then I add a prompt, the image on the right is completely different from the left image. How do I make them more of the same? I don't see a slider for deciding how close the output image should be to the original image. 

For example if I place a person on the left and they have brown hair, and I say "blonde hair" as the prompt, shouldn't the right image be exactly the same except with blonde hair? Instead, the image is changing completely.

I also keep getting an error "TypeError: string indices must be integers"

**Comment by Kyxsune** [comment]
I've found the error to be caused by an image (the default one) being in the mask option if crop is selected.

**Reply by dagfooyo** [reply]
How do you fix this?  I keep getting this error when I try to mask and I can't figure out what I'm doing wrong.  Crop is not selected for me

**Reply by Deadstarrr** [reply]
Yes, it seems you can fix this issue by going into the "mask" option and deleting the default image of the river. For some reason if you have crop selected and you delete the image, it still exists in the mask option, which seems to cause problems. I hope this is fixed soon.

**Reply by dagfooyo** [reply]
aha!  That worked for me!  That's been frustrating me all day.  Thank you!

**Comment by slashedstar** [comment]
On img2img the Denoising Strenght slider controls how much the generated image will differ from the input image

>For example if I place a person on the left and they have brown hair, and I saw "blonde hair" as the prompt, shouldn't the right image be exactly the same except with blonde hair? Instead, the image is changing completely.

If you just type blonde you aren't specifying much, neither the sex of the person in the original image nor the clothes they were wearing, the pose or type of image they were in (headshot portrait? side view? photography or painting?), so it will just convert everything to a very generic blonde, try to imagine that you aren't actually using img2img and just txt2img, try to reacrete the original image with the prompt + the one thing you want to change, if you had a headshot portrait photography of a white woman with black curly hair wearing a pink bra you would tipe something like `headshot portrait, white woman, blonde curly hair, wearing pink brassiere` on the img2img prompt, you may even try playing with weights to the prompts like `headshot portrait, white woman: blonde curly hair:2 wearing pink brassiere` to emphasize the blonde hair part (this website helps alot with the weighting syntax: https://promptomania.com/stable-diffusion-prompt-builder/)

But this is very hard to do anyway, because to change the color of the hair you need a high denoising value (drastic change to the image), but you don't want to drastically change the rest of the image, the way to try to navigate around this is to go with a low denoising value, generate a lot of images and pick the one with the most desirable features and feed it back to the img2img, thats how I convert sketches to photographs and vice-versa.
